### ISSUE TITLE
called code side effect

### ISSUE THREAD

**Issue by RemiCardona** [PyCQA/pylint#3620]
Spin off from bug #3611 to reduce noise.

### Steps to reproduce
1. fresh virtualenv using python's `venv`
2. `pip install pylint flask pyserial`
3. `grut.py` at root of venv directory:
```python
import flask
import serial


def initialize():
    print(flask.Blueprint)
    #print(flask.Config.from_pyfile)
    #print(flask.Config.from_json)
    try:
        serial.Serial()
    except serial.SerialException as exc:
        if isinstance(exc.strerror, str) and "could not open port" in exc.strerror:
            print(exc)
            return
        raise
```
4.
```console
$ pylint -E grut.py
************ Module grut
grut.py:12:70: E1135: Value 'exc.strerror' doesn't support membership test (unsupported-membership-test)
```
5. in `grut.py`, uncomment the 2nd or 3rd `print()` instead of the 1st one
6. error goes away

### Open questions

1. I'm not sure what the "correct" pylint behavior should be. To me, `OSError.strerror` is a `str` so the membership test is a false negative. It could also be `None` but the `isinstance()` gets us in path where we know for sure `in` is going to be allowed.
2. The really tricky bug here is that the order of code analysis changes pylint output, which sounds like a real bug.

### Useful information (I think…)

1. `serial.SerialException` inherits from `IOError`
   - https://github.com/pyserial/pyserial/blob/v3.4/serial/serialutil.py#L90
2. `IOError` is an "alias" (as per the CPython code) of `OSError`
   - https://github.com/python/cpython/blob/v3.7.7/Objects/exceptions.c#L2590
3. the flask code which gets pylint confused reraises `IOError` exceptions with a modified `strerror` attribute:
   - https://github.com/pallets/flask/blob/1.1.2/src/flask/config.py#L136 (`from_pyfile`)
   - https://github.com/pallets/flask/blob/1.1.2/src/flask/config.py#L200 (`from_json`)
4. the link with bug #3611 was that I found this bug through different module names on the command line, managed to narrow it down to 2 files and then reduced it to a single test case file

### pylint --version output
```console
$ pylint --version
pylint 2.5.2
astroid 2.4.1
Python 3.7.7 (default, Apr 22 2020, 23:22:50) 
[GCC 9.3.0]
```

**Comment by Pierre-Sassoulas** [comment]
I can reproduce this, but I think it's because in `flask.Config.from_pyfile` there is `            e.strerror = "Unable to load configuration file (%s)" % e.strerror` so astroid start inferring that `strerror` is a `str`. The thing is we infer thing based on code that is available, so I'm not sure it's a bug.
